app/models/user.rb

```diff
@@ -7,7 +7,47 @@ class User < ActiveRecord::Base
          :timeoutable, :confirmable
   include DeviseTokenAuth::Concerns::User
 
+  attr_accessor :initiate_email_reconfirmation
+
   def error_details(field)
     errors.details[field].collect { |i| i[:error] }
   end
+
+  def email_changed?
+    initiate_email_reconfirmation.present?
+  end
+
+  # override from Devise so that we can add hook into Salesforce update
+  def self.confirm_by_token(confirmation_token)
```

Comment:
ah yes you're right. nothing with salesforce, but i believe the story asks for the user to confirm their password change via email see: https://www.pivotaltracker.com/story/show/127691397